presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/RowExpressionFormatter.java

```diff
@@ -123,12 +144,30 @@ public String visitConstant(ConstantExpression node, ConnectorSession session)
             }
 
             Type type = node.getType();
-            if (node.getType().getJavaType() == Block.class) {
+            if (type.getJavaType() == Block.class) {
                 Block block = (Block) value;
                 // TODO: format block
                 return format("[Block: position count: %s; size: %s bytes]", block.getPositionCount(), block.getRetainedSizeInBytes());
             }
-            return type.getDisplayName().toUpperCase() + " " + value.toString();
+
+            String valueString = "'" + value.toString().replace("'", "''") + "'";
+
+            if (VarbinaryType.isVarbinaryType(type)) {
+                return "X" + valueString;
```

Comment:
Why is varbinary treated differently rather than using `varbinary'xxx'`?

Reply:
It seems like using TypeConstructor on varbinary takes the UTF-8 encoded version of the string as the binary literal, instead of reading it like a hex string. For example, when I run `SELECT varbinary'abc123', x'abc123', varbinary'😎'` on a local Presto server, I get 
```
presto:sf1> SELECT varbinary'abc123', x'abc123', varbinary'😎';
       _col0       |  _col1   |    _col2    
-------------------+----------+-------------
 61 62 63 31 32 33 | ab c1 23 | f0 9f 98 8e 
```
I'm not sure if this is intentional behavior or not, but since varbinary literals are output as hex strings, I needed this workaround.

Reply:
I moved it into a function `LiteralInterpreter.evaluateToString()`, since turning the constant to a string requires the relevant `ConnectorSession` and `LiteralInterpreter.evaluate()` contains similar logic. Thoughts?

Reply:
Sorry I didn't pay close attention. If it's not possible to move this to `ConstantExpression`, let's keep the logic in `RowExpressionFormatter`. Thanks!